Website/Packages/datax-pipeline/src/modules/flowDefinition/flowSelectors.js

```diff
@@ -235,11 +235,12 @@ export const getEnableLocalOneBox = createSelector(
 // Validation - Info
 export const validateFlowInfo = createSelector(
     getFlowDisplayName,
+    getFlowDatabricksToken,
     validateInfo
 );
 
-function validateInfo(displayName) {
-    return displayName && displayName.trim() !== '';
+function validateInfo(displayName, databricksToken) {
+    return displayName && displayName.trim() !== '' && databricksToken && databricksToken.startsWith('secretscope://');
```

Comment:
How does a new flow experience look like? Should we save it first to generate a secretscope URI and then deploy button will be enabled?

Reply:
Yes. We would need to first save and only then the deploy button will be enabled

Reply:
Then this changes the user experience for a new flow. Can we just delete this  databricksToken.startsWith('secretscope://' from the conditions?

Reply:
And you need to check the Spark type since databricksToken is always empty/null for HDInsight here as well.

Reply:
Since user has to always save the token before they can deploy, we need the databricksToken.startsWith('secretscope://' condition. Added check for spark type so that it works for HDInsight scenario